modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -1436,8 +1435,11 @@ bool PngEncoder::writeanimation(const Animation& animation, const std::vector<in
         }
     }
 
+    compression_level = compression_level < 0 ? Z_BEST_SPEED : compression_level;
```

Comment:
Why not just set Z_BEST_SPEED in constructor and remove the line here?

Reply:
then https://github.com/opencv/opencv/blob/4.x/modules/imgcodecs/src/grfmt_png.cpp#L910-L920 will be meaningless without `compression_level = -1;` let me look later with a fresh mind